packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -228,18 +235,16 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
     onHide,
     popoverProps,
     multiple = false,
+    triggerProps,
     ...rest
   } = props;
 
   useTheme();
 
-  const [isOpen, setIsOpen] = React.useState(popoverProps?.visible ?? false);
-  React.useEffect(() => {
-    setIsOpen((open) => popoverProps?.visible ?? open);
-  }, [popoverProps]);
+  const [isOpenState, setIsOpen] = React.useState(false);
```

Comment:
Nitpick, but maybe change `setIsOpen` to `setIsOpenState`

Reply:
This is done a bit intentionally because `isOpen` is defined right below it, so `isOpenState` is not used anywhere else in the code